.github/workflows/standard_spa_pr.yml

```diff
@@ -25,45 +46,31 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
-  cache-yarn:
-    name: Cache Yarn
-    runs-on: jupiterone-dev
-    steps:
-      - name: Checkout
-        uses: actions/checkout@v2
-      - name: Cache yarn dependencies
-        uses: actions/cache@v2
-        id: yarn-cache
-        with:
-          path: |
-            ${{ steps.yarn-cache-dir-path.outputs.dir }}
-            **/node_modules
-          key: cache-yarn-${{ hashFiles('**/yarn.lock') }}
-      - name: Install Dependencies
-        if: steps.yarn-cache.outputs.cache-hit != 'true'
-        # Eventually we want to flip to npm and use `npm ci` here
-        run: |
-          echo "//registry.npmjs.org/:_authToken=${{ secrets.NPM_AUTH_TOKEN }}" > .npmrc
-          yarn --frozen-lockfile
-
   validate:
     name: Validate
-    runs-on: jupiterone-dev
-    needs: cache-yarn
+    runs-on: ${{ inputs.runs_on }}
+    timeout-minutes: 15
+    if: ${{ inputs.use_validate }}
     steps:
-      - name: Checkout
-        uses: actions/checkout@v2
-      - name: Fetch Yarn Cache
-        uses: actions/cache@v2
-        id: yarn-cache
+      - uses: actions/checkout@v3
+      # This should be part of the runner
+      ## Install yarn START
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 16
+        if: ${{ inputs.install_yarn }}
+      - run: npm install -g yarn
+        if: ${{ inputs.install_yarn }}
+      ## Install yarn STOP
+      - uses: actions/setup-node@v3
         with:
-          path: |
-            ${{ steps.yarn-cache-dir-path.outputs.dir }}
-            **/node_modules
-          key: cache-yarn-${{ hashFiles('**/yarn.lock') }}
-      - name: Validating
-        # By design all SPAs are required to validate
-        # This should include all required CI checks
+          node-version: 16
+          cache: "yarn"
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+      - run: yarn --frozen-lockfile --ignore-optional
```

Comment:
Yarn 1 does not have a way to disable build scripts which is a bit scary. We have gotten away with this for now but we should continue to push to a full flip to GH actions and then NPM for better a security posture.

Reply:
Agreed and good to call out

Reply:
Is the concern with build scripts something coming from Security? I'm just hoping that we can identify when we're solving for problems that don't exist yet.